bench/BINARY_SIZE.md

```diff
@@ -16,9 +16,9 @@ The programs and their tests are located in [/tests/bench](https://github.com/co
 
 Solana version: 1.18.17
 
-| Program | Binary Size | -                     |
-| ------- | ----------- | --------------------- |
-| bench   | 787,968     | 🟢 **-3,040 (0.38%)** |
+| Program | Binary Size | -                        |
+| ------- | ----------- | ------------------------ |
+| bench   | 1,096,096   | 🔴 **+305,088 (38.57%)** |
```

Comment:
I was planning to merge this today, but the bench results show ~38% increase in binary size with this change. Always using closures for `init` constraints might not be what we want. Decreasing stack usage is certainly more important than binary size, but the difference is still massive.

Thoughts?

Reply:
I would kinda expect this might happen due to applying `inline(never)`: Closures benefit a lot from being able to inline and remove any redundant logic inside them.

Reply:
Thanks! I removed the `inline(never)`s, but it had very little effect on the binary size (+300k rather than +305k).

What we want is to inline them as long as the `try_accounts` function doesn't use more than 4096 bytes of stack. Not sure if there is a simple way to make the compiler care about the stack limit when deciding whether to inline or not.

Reply:
if inline had no major contribution in the 38% increase, what else could have caused such a difference ?

Reply:
The bench program is a rather unusual program that has 90 `init`s which skews the results. The binary size difference won't be as bad for regular programs.

I think we can get this in as is, since decreasing stack usage of `try_accounts` is very valuable currently (at least until we get [dynamic stack frames](https://github.com/solana-foundation/solana-improvement-documents/pull/166)).